tests/test_main.py

```diff
@@ -1976,6 +1989,22 @@ class Model(BaseModel):
     assert repr(m) == "Model(x={'one': 1, 'two': 2})"
 
 
+@pytest.mark.skipif(sys.version_info < (3, 7), reason='generic classes need 3.7')
+def test_typing_non_coercion_of_dict_subclasses():
+    KT = TypeVar('KT')
+    VT = TypeVar('VT')
+
+    class MyDict(Dict[KT, VT]):
+        def __repr__(self):
+            return f'MyDict({super().__repr__()})'
+
+    class Model(BaseModel):
+        a: MyDict
+        b: MyDict[str, int]
```

Comment:
```suggestion
        b: MyDict[str, int]
        c: Dict[str, int]
        d: Mapping[str, int]
```

What happens to these types with the input being a `MyDict`?

Reply:
for `c` coercion is done (as currently expected with pydantic) and for `d` no coercion is done. I added a test